### ISSUE TITLE
Permanently noisy world after certain visual effects

### ISSUE THREAD

**Issue by Paril** [NVIDIA/Q2RTX#226]
When certain temporary visual effects are used, sometimes the result is that that area of the world becomes permanently noisy.

Initially I was able to repro this with a material on a TE_LIGHTNING bolt, but I couldn't get this to happen again with the latest versions. However, I found that RF_BEAM entities do repro the issue. Some mods use RF_BEAMs as replacements for blaster bolts, for instance, but I don't know if they're able to reproduce this on a large scale.

A way to reproduce this is to shim this code above the call to `prepare_entities` on line 2905:

```
	// TEST
	// debug bones
	int x = Cvar_Get("pt_test_debug", "0", 0)->integer;
	for (size_t i = 0; i < x; i++) {
		vec3_t r = { 40 + (crand() * 40), crand() * 40, crand() * 40 };
		particle_t *pr = &fd->particles[fd->num_particles++];
		memset(pr, 0, sizeof(*pr));
		pr->color = -1;
		pr->brightness = 1;
		pr->radius = 1.0f;
		pr->rgba = (color_t) { .u8 = { 255, 0, 0, 255 } };
		pr->alpha = 1.f;
		VectorAdd(cl.playerEntityOrigin, r, pr->origin);

		entity_t *b = &fd->entities[fd->num_entities++];
		memset(b, 0, sizeof(*b));
		b->skinnum = -1;
		b->rgba = (color_t) { .u8 = { 0, 0, 255, 255 } };
		b->alpha = 0.30f;
		b->flags = RF_TRANSLUCENT | RF_BEAM;
		b->frame = 8;
		VectorCopy(pr->origin, b->origin);
		vec3_t r2 = { 40 + (crand() * 40), crand() * 40, crand() * 40 };
		VectorAdd(pr->origin, r2, b->oldorigin);
	}
	// TEST
```

this produces a fake but similar output to what I'm doing with my bone debugger. Load up q2dm1, set `pt_test_debug` to a number (40 is what I was testing with, but various numbers will work), then set it back to 0. Notice now that the area you are in is permanently noisy even though there are no more lasers lighting the scene.

**Comment by res2k** [comment]
> on line 2905:

Not sure this number is correct on current master; can you provide some context (lines before/after)?

**Comment by Paril** [comment]
Directly before the call to "prepare_entities". There's only one call to it, so it should be easy to find. It's 2828 on master I think?

Here's a video of the issue, but it may not be easy to see from a video. It's most noticeable in-game. https://streamable.com/736pzz

**Comment by res2k** [comment]
Ok, I can see, not in the video but in the actual game.
It's quite subtle. Not really "noisy", more of a permanent shimmer that's added.
I suspect it's less visible with higher framerate, but it can be seen better with lower resolution scale.
When disabling the denoiser the difference is more apparent - they are quite a few more "dark" pixels afterwards.
So maybe something affects the directions chosen for light sampling or so?

**Comment by Paril** [comment]
Yeah, a shimmer is a good way of describing it. It creates ghosting behind everything too, which would kind of make sense if it's direction vector related.

**Comment by apanteleev** [comment]
The project is discontinued; no new features or fixes will be implemented here. Closing.
